demo/state.h

```diff
@@ -78,6 +98,52 @@ struct State {
                                   HTTPResponseCode::OK,
                                   "image/svg+xml");
   }
+
+  void DemoRequest(Request r) {
+    if (r.http.Method() == "POST") {
+      // TODO(dkorolev): This should get simpler once Bricks 1.0 is out, the `.http.` will go away.
+      if (!r.http.HasBody()) {
+        points.emplace_back(atof(r.url.query["x"].c_str()),
+                            atof(r.url.query["y"].c_str()),
+                            !!atoi(r.url.query["label"].c_str()));
+      } else {
+        // Note: this will throw on an invalid JSON. TODO(dkorolev): Think of it further :-)
```

Comment:
Good point about the framwork.

I'll edit the comment instead of removing it :)

Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783


On Mon, Feb 16, 2015 at 6:37 PM, John Babak <notifications@github.com>
wrote:

> In demo/state.h
> <https://github.com/KnowSheet/KnowSheet/pull/8#discussion_r24789110>:
>
> > @@ -78,6 +98,52 @@ struct State {
> >                                    HTTPResponseCode::OK,
> >                                    "image/svg+xml");
> >    }
> > +
> > +  void DemoRequest(Request r) {
> > +    if (r.http.Method() == "POST") {
> > +      // TODO(dkorolev): This should get simpler once Bricks 1.0 is out, the `.http.` will go away.
> > +      if (!r.http.HasBody()) {
> > +        points.emplace_back(atof(r.url.query["x"].c_str()),
> > +                            atof(r.url.query["y"].c_str()),
> > +                            !!atoi(r.url.query["label"].c_str()));
> > +      } else {
> > +        // Note: this will throw on an invalid JSON. TODO(dkorolev): Think of it further :-)
>
> Wow :) That was just a thought on how error handling should be implemented
> in the framework, not in the demo. OK, you could likely remove the comment,
> too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/KnowSheet/pull/8/files#r24789110>.
>